csrc/scheduler/transpose.cpp

```diff
@@ -227,34 +227,31 @@ class DomainMap : public pointwise_utils::DomainMap {
         root_dim,
         " in tensor ",
         tv);
-    auto replay_exprs = StmtSort::getExprsBetween(
+    std::vector<Expr*> replay_exprs = StmtSort::getExprsBetween(
         {mapped_id}, {tv->getLoopDomain().begin(), tv->getLoopDomain().end()});
     // Project the root id to loop id. Similar to projectIdToRFactor.
-    for (auto expr : replay_exprs) {
-      if (expr->isA<Split>()) {
-        // Split with factor one is not supposed to be here, reshape would map
-        // this to a broadcast. This is a conservative assert, we can relaxed it
-        // and support with mapping it to outer.
-        NVF_ERROR(
-            !expr->as<Split>()->factor()->isOneInt(),
-            "split with factor one is supposed to be translated to broadcast by reshape");
```

Comment:
This assumption no longer holds for DID split and therefore is removed. Other changes in this file are cosmetic.

Reply:
When does the DID split use split by a factor of one?

Reply:
When there's only one gpu e.g. running test-multidevice with mpirun np 1. Valid but not so interesting

Reply:
I see. I think we should do:

```
if (split->inner()->extent()->isOneInt() && !split->outer->extent()->isOneInt()) {
  mapped_id = split->outer();
} else {
  mapped_id = split->inner();
}
```

Here, we are traversing down to the innermost loop ID, ignoring broadcast IDs.